flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCLookupFunction.java

```diff
@@ -101,7 +113,8 @@ public JDBCLookupFunction(
 		this.maxRetryTimes = lookupOptions.getMaxRetryTimes();
 		this.keySqlTypes = Arrays.stream(keyTypes).mapToInt(JDBCTypeUtil::typeInformationToSqlType).toArray();
 		this.outputSqlTypes = Arrays.stream(fieldTypes).mapToInt(JDBCTypeUtil::typeInformationToSqlType).toArray();
-		this.query = options.getDialect().getSelectFromStatement(
+		this.nonNullableQuery = options.getDialect().getSelectFromStatement(options.getTableName(), fieldNames, keyNames);
+		this.nullableQuery = options.getDialect().getSelectNotDistinctFromStatement(
```

Comment:
If we have a consensus on the new methods of `SqlDialect`. we can call the field `nullSafeQuery` and `nonNullableQuery` -> `nullUnsafeQuery`.